apps/ledger-live-mobile/src/screens/ClaimNft/ClaimNftQrScan.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useCallback, useState, useEffect } from "react";
+import { Linking } from "react-native";
+import { SafeAreaView } from "react-native-safe-area-context";
+import { Button, Flex, Icons, Text, Box } from "@ledgerhq/native-ui";
+import { useTranslation } from "react-i18next";
+import { CameraType } from "expo-camera/build/Camera.types";
+import { Camera } from "expo-camera";
+import { BarCodeScanner } from "expo-barcode-scanner";
+import { Svg, Defs, Rect, Mask, Circle } from "react-native-svg";
+import { urls } from "../../config/urls";
+
+const WrappedSvg = () => (
+  <Svg height="100%" width="100%" viewBox="0 0 100 100">
+    <Defs>
+      <Mask id="mask" x="0" y="0" height="100%" width="100%">
+        <Rect height="100%" width="100%" fill="#fff" />
+        <Circle r="45" cx="50" cy="50" />
+      </Mask>
+    </Defs>
+    <Rect
+      height="100%"
+      width="100%"
+      fill="rgba(0, 0, 0, 0.5)"
+      mask="url(#mask)"
+      fill-opacity="0"
+    />
+  </Svg>
+);
+
+const ClaimNftQrScan = () => {
+  const { t } = useTranslation();
+  const [permission, requestPermission] = Camera.useCameraPermissions();
+
+  const handleBarCodeScanned = useCallback(({ data }) => {
+    try {
+      const url = new URL(data);
+      const { hostname } = url;
+      const code = url.href.substring(url.href.lastIndexOf("/") + 1);
+      const deeplink =
+        hostname === "staging.claim.ledger.com"
+          ? urls.discover.linkDropStaging + "?redirectToOnboarding=true&code="
+          : urls.discover.linkDrop + "?redirectToOnboarding=true&code=";
+      Linking.openURL(deeplink + code);
```

Comment:
It would be cleaner to use URLParams to build the URL rather than doing string concatenation.
I'm not actually sure this API is available in react native so I'll let you try.